### PR TITLE
Sanitize column names on data import

### PR DESCRIPTION
**Pull request by wverastegui**
This PR introduces features to standardize the 'retention time' and 'retention index' column names, aligns the unit tests with these changes, and enhances code readability by the application of docstrings and type annotations.

Closes https://github.com/RECETOX/RIAssigner/issues/98 and https://github.com/RECETOX/RIAssigner/issues/93

### CODE REVIEW

File: RIAssigner/data/Data.py [hunk 1]
```diff
@@ -29,38 +29,65 @@ def is_valid(value: Union[RetentionTimeType, RetentionIndexType]) -> bool:
...
 
         Returns:
-            List[str]: List of supported retention index keys.
+            List[str]:  A list of possible keys to get retention indices.
         """
-        return cls._ri_possible_keys.copy()
+        keys_conversions = load_known_key_conversions()
+        ri_key_converted = [ key for key, value in keys_conversions.items() if "retention_index" == value]
+        ri_key_converted.append("retention_index")
+        return ri_key_converted
```
Comment (hechth) [inline RIAssigner/data/Data.py:90]: This should still be adapted to be in line with how the retention time keys are initialzied and obtained.

File: tests/test_data_Data.py
Comment (hechth) [inline tests/test_data_Data.py:1]: Please add the tests for the retention index similarly to how the behaviour for the retention time is tested.

**Review by hechth — CHANGES_REQUESTED** [review]
Please add the missing tests for the RI reading and adding values to it behaviour.

File: RIAssigner/data/Data.py (hunk 1 above)
Comment (wverastegui) [reply RIAssigner/data/Data.py]: Addressed [5ff2049](https://github.com/RECETOX/RIAssigner/pull/118/commits/5ff2049bc559c853cead7029770afef13430d25a)

File: tests/test_data_Data.py
Comment (wverastegui) [reply tests/test_data_Data.py:1]: Missing test cases added [3a2055b](https://github.com/RECETOX/RIAssigner/pull/118/commits/3a2055be5c597f0fd7e9cde9feebab4dfd14ed22)

**Review by hechth — APPROVED** [review]

**Pull request merged**